pkg/sql/pgwire/conn.go

```diff
@@ -555,8 +555,7 @@ func (c *conn) serveImpl(
 				// protocol and encounters an error, everything until the next sync
 				// message has to be skipped. See:
 				// https://www.postgresql.org/docs/current/10/protocol-flow.html#PROTOCOL-FLOW-EXT-QUERY
-
-				return false, isSimpleQuery, c.stmtBuf.Push(ctx, sql.Sync{})
+				return false, isSimpleQuery, c.stmtBuf.Push(ctx, sql.Sync{CameFromClient: true})
```

Comment:
Ditto

Reply:
and similarly, in this case, we are sending Sync because there _is_ a ClientMsgSync, so this flag should be set